Rewriting to actually work

fill1890 opened the pull request:
Not actually functional yet. Initially adding a queue class for internal transmission use

fill1890 committed:
Add a queue

Not yet functional or used

dawnniie reviewed (commented):
lgtm! documentation kinda cringe though

fill1890 committed:
Squashed some bugs

dawnniie committed:
greatly improve documentation

fill1890 committed:
Implement a stream

Add a stream to replace the queue. Uses events and is scoped to only be
a stream

Removes the old queue as it will no longer be necessary and I don't like
it

fill1890 committed:
Implement a keepalive module

Implements a keepalive module to send keepalive messages and keep track
of missing responses.

dawnniie commented:
https://www.youtube.com/watch?v=Fb7N-JtQWGI

fill1890 commented:
Please keep discussion relevant

On Sat, 11 Sep 2021 at 6:54 pm, eeehh ***@***.***> wrote:

> https://www.youtube.com/watch?v=Fb7N-JtQWGI
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fill1890/js-milight-v6/pull/4#issuecomment-917370544>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVER6DXXG6G7GZPCV3U2ALUBMKNTANCNFSM5DYOAL6Q>
> .
>
>
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

fill1890 committed:
Add a receiver module

Adds a receiver module to listen for incoming messages
